### PR TITLE
Update homework03.cc

### PR DESCRIPTION
**Pull request by AJIEIIIA**
implementation

### CODE REVIEW

File: course5/homework03/homework03.cc [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+}
+
+void print(int value)
+{
+	std::printf("%d", value);
+}
+
+void print(float value, uint precision = 0)
+{
+    	std::string format = std::string("%.") + std::to_string(precision).c_str() + std::string("f");
```
Comment (maciekgajewski) [inline course5/homework03/homework03.cc:22]: Consider ```%.*f```

**Review by maciekgajewski — COMMENTED** [review]
Good

**Comment by AJIEIIIA** [comment]
Maciek, why does my program coredump at line 80?

File: course5/homework04/homework04.cc [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+};
+
+int main(int argc, char** argv)
+{
+
+        Stream Cout;
+
+        // why: double free or corruption (top) ?
+        if (argc > 1)
+                Cout = Stream(argv[1]);
```
Comment (vargad) [inline course5/homework04/homework04.cc:80]: Here a temporary Stream object is created, which opens the file named argv[1]. The object get copied to Cout object, effectivly coping the FILE pointer to Cout. After the if statement the temporary Stream object gets destroyed, so it closes the file making the pointer invalid, while the Cout object still has pointer to it.

**Review by maciekgajewski — CHANGES_REQUESTED** [review]
Please, deliver every homework in a separate pull request

**Pull request closed without merging**